lib/domain/entities/track_point.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:movna/domain/entities/location.dart';
+
+part 'track_point.freezed.dart';
+
+/// A point crossed by user at an instant, with associated data.
+@freezed
+class TrackPoint with _$TrackPoint {
+  const TrackPoint._();
+
+  const factory TrackPoint({
+    /// Location with movement data and timestamp.
+    Location? location,
```

Comment:
why can this be null?

Reply:
also think that the dateTime is more relevant as a property of the TrackPoint rather than the location

Reply:
Yes agree